### PR TITLE
Add support to empty lists, remove Discriminator tag from json output

### PR DESCRIPTION
**Pull request by thisAAY**
Here's a revised version of the GitHub Pull Request description:

---

When attempting to decode a List from an empty JSON list as shown below:
```json
{
  "links": {
    "self": "https://root.link.com"
  },
  "data": []
}
```

An error is generated. Previously, if the list was empty, the type was inferred using the `Model` type, such as `XItem`. This approach is incorrect. Instead, we should directly use the `Data` type. For instance, in the case of the `Person` example, we should utilize the `Person` class rather than `PersonList`.

---

I've incorporated additional tests to address scenarios involving Meta and links. I'll continue to add tests as I identify new edge cases. At present, our tests focus on decoding and encoding. I plan to extend testing to include the generation process in the future.

### CODE REVIEW

File: core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt [hunk 1]
```diff
@@ -73,12 +78,31 @@ class CommonDiscriminator(private val discriminator: String) : Discriminator {
 
     private fun removeDiscriminatorEntry(jsonObject: JsonObject): JsonObject {
         return jsonObject.entries
-            .toMutableSet()
+            .toMutableList()
```
Comment (Aksi0m) [inline core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt:81]: Did not get the reason why set did not work here? Can you explain in more detail?

**Review by Aksi0m — APPROVED** [review]
Looks good. I have one question.

File: core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt (hunk 1 above)
Comment (thisAAY) [reply core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt:81]: I modified the function to handle the extraction and reinsertion of meta and links without changing their positions. The use of `Set` was avoided to maintain the original order and prevent a non-standard JSON format, which would have otherwise pushed meta and links to the end of the document and required updates to our test JSON files.

https://github.com/infinum/kotlin-jsonapix/blob/94de42b88b13e21fee5d79e996ccdcc6d14fd84f/core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt#L96-L105
